### ISSUE TITLE
How to get candidate solutions from each iteration?

### ISSUE THREAD

**Issue by BruthYU** [Evolutionary-Intelligence/pypop#72]
Thanks for the excellent work :partying_face: 
1. I'm looking for black-box optimization algorithms to perform prompt tuning on my neural network, which requires candidate solutions from each iteration. 
2. Following the “ask-tell” form of [pycma](https://github.com/CMA-ES/pycma), the pseudo-code is shown as below:
```
while not es.stop():
    solutions = es.ask()
    fitness = [cma.ff.rosen(s) for s in solutions]
    es.tell(solutions, fitness)
```
3. **Question**: How can I obtain the solutions from each iteration to calculate new fitness (e.g. loss functions)?

**Comment by Evolutionary-Intelligence** [comment]
Thanks for your interest. Currently this libaray does not return the solution from each iteration *by default*. We only return the **solution value** for saving memory. For large-scale black-box optimization, typically returning the solutions from each iteration is memory-expensive ().

However, it is possible to do this via slightly modifying the code. 
Here is a related example: https://github.com/Evolutionary-Intelligence/pypop/blob/main/pypop7/optimizers/es/_repeat_fmaes.py#L74

In fact, you only need to add a variable to record it (I think this is a very simple coding task since all the optimizer class are OOP).

**Comment by Evolutionary-Intelligence** [comment]
If you still have questions, you can write a simple demo. And I can help to add them. Thanks for your interest again.

**Comment by Evolutionary-Intelligence** [comment]
@twoletters : I am very sorry that I see this report until NOW (August 20, 2023 not Feb 1, 2023). We will add this in the near future version (from pypop7 0.0.72). Thanks for your attention.

**Comment by Evolutionary-Intelligence** [comment]
I am very sorry again that we will add this in the near future version (from 0.0.77 rather 0.0.72). This is because we are busy to speedup some optimizers owing to their *impefect* designs. Thanks for your attention.
